tests/deepsparse/loggers/metric_functions/template_functions/test_template_functions.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+from deepsparse import Pipeline
+from tests.utils import mock_engine
+
+
+yaml_config = """
+loggers:
+    list_logger:
+        path: tests/deepsparse/loggers/helpers.py:ListLogger
+data_logging:
+    predefined:
+    - func: image_classification
+      frequency: 1
```

Comment:
It's not clear from the current predefined block what is being passed into the functions (is it pipeline inputs, outputs, everything?), and it's also a different paradigm than we are using for other functions. This means when documenting this feature we have to have a special section for specifying predefined things that work differently from the rest.

Additionally, since this is an inconsistency with the rest of the UX, as a user I might wonder if func/frequency mean the same thing here as they do for other functions? This will increase mental burden when writing yaml configs, as its adding a second set of rules a user has to know.

That being said I suggest the following UX:
```suggestion
    pipeline_inputs:
    - func: builtins.image_classification
      frequency: 1
```
(alternatively can name it `predefined.image_classification`)

I think this formulation would be more clear to user, and also more consistent with our existing setup. Further it would allow more fine grained control on what exactly is being logged in a way that is clear just from the yaml file.

Reply:
Thanks for that @corey-nm .I agree that your proposal adds a mental burden for users, but I think it's manageable with proper documentation. The alternative "convention" gives a different meaning to a template identifier, which can be confusing but eventually may be understandable. However, what I don't like about the proposal is that users still need to think about different target identifiers, such as pipeline_inputs, pipeline_outputs, etc. My main goal was to have one pathway that loads all functions for all identifiers at once. To summarize, your proposal reduces some of the mental burden for users, but it also requires more explicit configs. I see there is a balance to strike here, I would like to hear Ben's opinion on this as well.

Reply:
Predefined was selected to as a catch all since the built ins cover multiple pipeline values. Could potentially choose a different name than predefined, but will need some catch all value as @dbogunowicz mentioned

Reply:
@dbogunowicz @bfineran Can this be handled in a block separate from data_logging then? For example:

```yaml
inherits_from:
  - builtins.image_classification
data_logging:
  whatever: {...}
  can_be_anything: {...}
```

The nice thing about this is it is explicitly different from the rest of data_logging, and users would know it has different rules. It becomes very easy to document this additional block when writing the builtins, and since its just a list of strings, its easy to use.

The other thing is this allows us to inherit more things than just logging in the future if we wanted to (e.g. hyperparameters).

Reply:
In that case, we need to maintain and process additional "config", I remain sceptical and not convinced :)

Reply:
Ask chatGPT to explain the existing config and the one I posted and see which one it explains better. If it explains both equally well then let's go with the existing implementation, but I'm guessing it won't know what it does 😀

Reply:
And also, I'd be willing to pay some slight configuration parsing complexity for better UX 🤷

Reply:
<img width="802" alt="image" src="https://user-images.githubusercontent.com/97082108/214606704-3f5f2ad8-b0c6-4d6b-bd23-c6cdd24a51dd.png">
I think you are collaborating

Reply:
😉 i knew i liked chatGPT

Reply:
the roman orators call it argument ad robotum

Reply:
🤣

Reply:
this LGTM, want to be cautious of adding too many args in the future, but prefer this over the more "hidden" functionality we currently have.

"inherits_from" sounds a bit vague for the actual purpose. thoughts on `add_buildins` or `function_groups` as a key instead?